src/custom/state/orders/hooks.ts

```diff
@@ -57,6 +59,31 @@ export const useOrder = ({ id, chainId }: GetRemoveOrderParams): Order | undefin
   })
 }
 
+// used to extract Order.summary before showing popup
+// TODO: put the whole logic inside Popup middleware
+export const useFindOrderById = ({ chainId }: GetOrdersParams): GetOrderByIdCallback => {
+  const state = useSelector<AppState, OrdersState[ChainId] | undefined>(state => chainId && state.orders?.[chainId])
+
+  // stable ref, so we don't recreate the function
+  const stateRef = useRef(state)
```

Comment:
im lost here, i thought useSelector did this for us

Reply:
oh nvm it's for the useCallback I see.

Reply:
It does return an order, there's even `useOrder` hook
But I want to lookup multiple orders lazily, after we fetch events
If we `useSelector|useOrder`, the order it returns will need to be a dependency of the `useEffect` where fetching takes place, so we'll need to manage state changing there, while we juggle events from a previous fetch

With use Ref we always have a ref to current orders and can lazily search through it

BUT I'll remove the whole thing when I make Popup middleware